autonomy/deploy/base.py

```diff
@@ -216,6 +220,60 @@ def read_keys(self, keys_file: Path) -> None:
 
         self._keys = keys
 
+    @staticmethod
+    def _try_update_safe_contract_address(
+        address: str,
+        override: Dict,
+        skill_id: PublicId,
+        has_multiple_overrides: bool = False,
+    ) -> None:
+        """Try update the `safe_contract_address` parameter"""
+
+        try:
+            if not has_multiple_overrides:
+                override["models"]["params"]["args"]["setup"][
+                    "safe_contract_address"
+                ] = [
+                    address,
+                ]
+                return
+
+            for agent_idx in override:
+                override[agent_idx]["models"]["params"]["args"]["setup"][
+                    "safe_contract_address"
+                ] = [address]
+        except KeyError:
+            logging.warning(
+                f"Could not update the `safe_contract_address` parameter for {skill_id}; "
+                "Configuration does not contain the json path to `safe_contract_address` parameter"
+            )
+
+    def try_update_multisig_address(self, address: str) -> None:
+        """Try and update multisig address if `safe_contract_address` parameter is defined."""
```

Comment:
So now we update the in memory skill overrides at the runtime and service config stays unaffected.